### DISCUSSION TITLE
v3.0.0-rc.11

### DISCUSSION THREAD

**Discussion by pi0** [nuxt/nuxt#16037] [Announcements]
[**🗒️ Full Release Notes**](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.11)

## 🚀 How to Upgrade

> **Note**
> Make sure to recreate the lock file in the project in case of any issues after the upgrade.

- **Automated:** `npx nuxi@latest upgrade --force`
- **Manual:** Bump `nuxt` dependency to `3.0.0-rc.11` and then use `npx nuxi@latest cleanup` to cleanup any local caches

### Full Static Enhancements

We have introduced Full-Static mode payload extraction in RC.10. Many of the issues from the initial implementation are resolved with this release thanks to your amazing feedback! Notably for SPA routes and state that is now in the initial state.

🧪 We understand that there might be still issues with the new implementation. Please report if spotted any. You can use new `experimental.payloadExtraction: false` flag in `nuxt.config` to opt-out as well.

### IPv6 and HTTPS support for `nuxi dev` and vite

Nuxi CLI and [unjs/listhen](https://github.com/unjs/listhen) are improved and now support `--https` flag and ipv6 hosts out of the box with an auto-generated certificate. You can use `--ssl-cert` and `--ssl-key` to provide own generated SSL certificates with [mkcert](https://github.com/FiloSottile/mkcert) for example as well.

Issues with vite HMR and vite-node should be resolved as well. If you were previously using `NODE_TLS_REJECT_UNAUTHORIZED` or custom `vite.server.hmr` options for a workaround, you can try to remove them.

**Note:** If you see something like `http://[::]:3000/` when running `nuxi preview`, it is all normal! The New IPv6 URL works in all modern browsers and is also backward compatible with IPv4 interfaces. If for some reason encountered any issues, try setting `HOST` to `0.0.0.0` to disable the IPv6 listener.

### Nitro Improvements

Nitro is the server engine for Nuxt 3. We had landed several fixes in [0.5.2 and 0.5.3](https://github.com/unjs/nitro/blob/main/CHANGELOG.md#v053) versions improving stability and bug fixes.

--- 

[**🗒️ Full Release Notes**](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.11)

**Comment by productdevbook** [comment]
thank you

**Comment by FileSoft-us** [comment]
<details>
  <summary>logs</summary>
11:59:50 PM: [error] [vite:vue] Transform failed with 2 errors:
11:59:50 PM: /opt/build/repo/node_modules/nuxt-icon/dist/runtime/Icon.vue:8:0: ERROR: Transforming const to the configured target environment ("ES5") is not supported yet
11:59:50 PM: /opt/build/repo/node_modules/nuxt-icon/dist/runtime/Icon.vue:10:2: ERROR: Transforming const to the configured target environment ("ES5") is not supported yet
11:59:50 PM: file: /opt/build/repo/node_modules/nuxt-icon/dist/runtime/Icon.vue:8:0
11:59:50 PM: 
11:59:50 PM: Transforming const to the configured target environment ("ES5") is not supported yet
11:59:50 PM: 5  |  import "/opt/build/repo/node_modules/nuxt-icon/dist/runtime/Icon.vue?vue&type=style&index=0&scoped=1ab6d6c7&lang.css"
11:59:50 PM: 6  |
11:59:50 PM: 7  |  import { useSSRContext as __vite_useSSRContext } from 'vue'
11:59:50 PM:    |                                                              ^
11:59:50 PM: 8  |  const _sfc_setup = _sfc_main.setup
11:59:50 PM:    |  ^
11:59:50 PM: 9  |  _sfc_main.setup = (props, ctx) => {
11:59:50 PM: Transforming const to the configured target environment ("ES5") is not supported yet
11:59:50 PM: 8  |  const _sfc_setup = _sfc_main.setup
11:59:50 PM: 9  |  _sfc_main.setup = (props, ctx) => {
11:59:50 PM: 10 |    const ssrContext = __vite_useSSRContext()
11:59:52 PM: Creating deploy upload records
11:59:52 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
11:59:50 PM:    |    ^
11:59:50 PM: 11 |    ;(ssrContext.modules || (ssrContext.modules = new Set())).add("node_modules/nuxt-icon/dist/runtime/Icon.vue")
11:59:50 PM: 12 |    return _sfc_setup ? _sfc_setup(props, ctx) : undefined
11:59:50 PM: 
</details>

**Reply by pi0** [reply]
Thanks for sharing logs. i've installed RC.11 with fresh lock file and latest `nuxt-icon`. Netlify build should be fixed with RC.11. Can you upgrade with `--force` and retry?

**Reply by FileSoft-us** [reply]
> Thanks for sharing logs. i've installed RC.11 with fresh lock file and latest `nuxt-icon`. Netlify build should be fixed with RC.11. Can you upgrade with `--force` and retry?

here's my tsconfig.json file...  what should to be fixed 

  "compilerOptions": {
    "target": "ES5",
    "module": "esnext",
    "moduleResolution": "node",
    "strict": false,
    "lib": ["dom", "dom.iterable", "esnext", "es6", "scripthost"],
    "declaration": false /* Generates corresponding '.d.ts' file. */,
    "sourceMap": true
  },

**Reply by pi0** [reply]
Normally `tsconfig` should look like [this](https://github.com/nuxt/starter/blob/v3/tsconfig.json), extending `./.nuxt/tsconfig.json` only. Would be nice if you can share a reproduction of your project 🙏🏼

**Reply by FileSoft-us** [reply]
yes there is no more error after disable "target": "ES5", in "compilerOptions"
but i have problem with all form reset with initial value 

let initForm = {
  name: '',
  desc: '',
  sort_order: 1,
  published: true,
}
let dForm = $ref({ ...initForm });

======
reset function 

dForm = Object.assign({}, initForm);

**Comment by Amirali-Yavari** [comment]
thanks dear nuxt team

**Comment by lucassimines** [comment]
Thanks 💚

**Comment by maxdzin** [comment]
Thanks a lot! 💚

**Comment by dolbex** [comment]
Much appreciated!!!

**Comment by TheDutchCoder** [comment]
Unfortunately this results in a whole bunch of fs errors for me, when I upgrade to RC11 from RC10.
Same issue when just running just `--https`

<details>
Nuxi 3.0.0-rc.11                                                                                                                                                                                                                          13:58:52
Nuxt 3.0.0-rc.11 with Nitro 0.5.4                                                                                                                                                                                                         13:58:52
                                                                                                                                                                                                                                          13:58:52
  > Local:    https://localhost:3000/
  > Network:  https://192.168.1.248:3000/

ℹ Using Tailwind CSS from ~/assets/css/tailwind.css                                                                                                                                                                      nuxt:tailwindcss 13:58:52

 ERROR  (node:28484) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.                                               13:58:53
(Use `node --trace-warnings ...` to show where the warning was created)

ℹ Tailwind Viewer: https://[::]:3000/_tailwind/                                                                                                                                                                          nuxt:tailwindcss 13:58:54
✔ Nitro built in 285 ms                                                                                                                                                                                                             nitro 13:58:56
✘ [ERROR] No matching export in "node-modules-polyfills:fs" for import "existsSync"

    node_modules/local-pkg/dist/shared.mjs:31:9:
      31 │ import { existsSync, promises as fs2 } from "fs";
         ╵          ~~~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:fs" for import "promises"

    node_modules/local-pkg/dist/shared.mjs:31:21:
      31 │ import { existsSync, promises as fs2 } from "fs";
         ╵                      ~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:url" for import "fileURLToPath"

    node_modules/local-pkg/dist/shared.mjs:35:9:
      35 │ import { fileURLToPath as fileURLToPath2 } from "url";
         ╵          ~~~~~~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:fs" for import "promises"

    node_modules/local-pkg/dist/shared.mjs:40:13:
      40 │ import fs, { promises as fsPromises } from "fs";
         ╵              ~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:url" for import "fileURLToPath"

    node_modules/local-pkg/dist/shared.mjs:41:9:
      41 │ import { fileURLToPath } from "url";
         ╵          ~~~~~~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:fs" for import "existsSync"

    node_modules/local-pkg/index.mjs:2:9:
      2 │ import { existsSync, promises as fs, readFileSync } from 'fs'
        ╵          ~~~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:fs" for import "promises"

    node_modules/local-pkg/index.mjs:2:21:
      2 │ import { existsSync, promises as fs, readFileSync } from 'fs'
        ╵                      ~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:fs" for import "readFileSync"

    node_modules/local-pkg/index.mjs:2:37:
      2 │ import { existsSync, promises as fs, readFileSync } from 'fs'
        ╵                                      ~~~~~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:module" for import "createRequire"

    node_modules/local-pkg/index.mjs:3:9:
      3 │ import { createRequire } from 'module'
        ╵          ~~~~~~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:buffer" for import "default"

    node_modules/vitest/dist/vendor-index.2ae8040a.mjs:5:7:
      5 │ import require$$0$1 from 'buffer';
        ╵        ~~~~~~~~~~~~

✘ [ERROR] No matching export in "node-modules-polyfills:fs" for import "promises"

    node_modules/vitest/dist/chunk-runtime-chain.be610650.mjs:7:13:
      7 │ import fs, { promises } from 'fs';
        ╵              ~~~~~~~~


 ERROR  1:58:56 p.m. [vite] error while updating dependencies:                                                                                                                                                                            13:58:56
Error: Build failed with 11 errors:
node_modules/local-pkg/dist/shared.mjs:31:9: ERROR: No matching export in "node-modules-polyfills:fs" for import "existsSync"
node_modules/local-pkg/dist/shared.mjs:31:21: ERROR: No matching export in "node-modules-polyfills:fs" for import "promises"
node_modules/local-pkg/dist/shared.mjs:35:9: ERROR: No matching export in "node-modules-polyfills:url" for import "fileURLToPath"
node_modules/local-pkg/dist/shared.mjs:40:13: ERROR: No matching export in "node-modules-polyfills:fs" for import "promises"
node_modules/local-pkg/dist/shared.mjs:41:9: ERROR: No matching export in "node-modules-polyfills:url" for import "fileURLToPath"
...
    at failureErrorWithLog (/Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:1566:15)
    at /Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:1024:28
    at runOnEndCallbacks (/Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:1438:61)
    at buildResponseToResult (/Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:1022:7)
    at /Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:1134:14
    at responseCallbacks.<computed> (/Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:671:9)
    at handleIncomingPacket (/Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:726:9)
    at Socket.readFromStdout (/Users/reinierkaper/projects/remx/apps/nuxt-app/node_modules/esbuild/lib/main.js:647:7)
    at Socket.emit (node:events:527:28)
    at Socket.emit (node:domain:475:12)
</details>

**Reply by pi0** [reply]
Do you mind sharing a reproduction? 🙏🏼 (Also your Node.js version)

Tip: Try to recreate lockfile. It might be because of outdated dependencies as well.

**Reply by TheDutchCoder** [reply]
That'll be tricky (private repo). I'm currently trying to do a full clean (including npm cache) and see if that yields anything.

I'll keep you updated!

Edit: Node is 16.15.1

**Reply by TheDutchCoder** [reply]
Shoot, no effect. This is really weird. Everything working fine before the upgrade and now I can't even get it to work anymore.

I guess I'll just remove all deps until I find the issue. This might take a while :(

**Reply by pi0** [reply]
Feel free to DM me in discord in case need extra eyes to check private repo.

**Comment by cesswhite** [comment]
Amazing work Nuxt Team, thank you! ❤️

**Comment by davidhellmann** [comment]
How is it possible to add routes dynamically here?

```js
  nitro: {
    prerender: {
      routes: ['/']
    }
  },
  ```

Is it possible to call Axios here to get all routes from a cms? Or whats the way to go to fill this with routes?

**Reply by davidhellmann** [reply]
Found this: https://github.com/nuxt/framework/issues/4919#issuecomment-1124349857
Would be great to have stuff like this somewhere in the docs

**Comment by thunder87** [comment]
Thanks! Huge impact upgrading from rc9 🤩 Keep up the good work 💪

**Comment by JuanxCursed** [comment]
amaaaaaaaazing job guys, just can't wait for the stable version to use on production environments

**Reply by hadelnasr** [reply]
Does rc means that this is not a stable version yet?

**Reply by MarcusOhman91** [reply]
@hadelnasr  RC is not stable. Roadmap still says "summer 2022" for stable version,  new ETA for stable would be appreciated. I'm starting a new project very soon that is due to go to production in 3 months and I'm not sure whether to v3 or v2

**Reply by hadelnasr** [reply]
@MarcusOhman91 Can you share this Roadmap?

**Reply by AXYZE9** [reply]
@hadelnasr Here you go https://v3.nuxtjs.org/community/roadmap

**Reply by JuanxCursed** [reply]
"Summer 2022" ends tomorrow that "should" means that we will have a stable version tomorrow until 11:59 pm 😃 right @pi0?

**Reply by kosmeln** [reply]
See a comment made by @pi0 in a separate thread  https://github.com/nuxt/framework/discussions/7513#discussioncomment-3651081

**Reply by zhaoweizhong** [reply]
Now it became "Autumn 2022". https://v3.nuxtjs.org/community/roadmap/

**Reply by rahulkumarsingh73690** [reply]
I hope October

**Comment by JenuelDev** [comment]
the best framework!

**Comment by TuguldurJ** [comment]
Finally, we can compile without the NODE_TLS_REJECT_UNAUTHORIZED warning message. Thank you guys!

**Comment by SaeedP11** [comment]
Amazing work Nuxt, you saved my project, thanks

**Reply by procrates** [reply]
* Best full stack framework ever *

**Comment by buglavecz** [comment]
Nice job. Best vue framework ever. 🚀

**Comment by nurik-nuxt** [comment]
Could not resolve import "@babel/runtime/helpers/esm on building
How to solve?

**Comment by AXYZE9** [comment]
Great news and awesome work guys! I hope stable 3.0 is coming soon 😊

**Comment by F-NiX** [comment]


**Reply by dolbex** [reply]
@F-NiX 

Clearly you are not trying to be rude so please don't take my reply that way either. 

What you have to keep in mind is this simple fact: This is a passion project for a lot of people. They are building it with the hopes that it's something great. Maybe for money one way or the other but mainly just to build something great. At the end of the day none of this work guarantees anyone any money. 

You, on the other hand, are building a project for profit with a client that is entrusting that you are the professional that knows the right tech stack for their solution. You should probably never put a client that in the position of working from a bleeding edge technology. _Especially_ if that solution relies are solutions that aren't even built yet (Nuxt core, image, auth, etc). 

The good news is it may not be _that_ hard to move your code to Nuxt 2 Bridge. I'd much rather the Nuxt team take their time get it right and for them to find their own success while I piggyback on their hard work. I'm sure you feel the same.

**Comment by szulcus** [comment]
After accidentally updating from `Nuxi 3.0.0-rc.8` to `Nuxi 3.0.0-rc.11`, a lot of things stopped working for me. How can I go back to version `Nuxi 3.0.0-rc.11`?
I am using Nuxt 2 with Bridge engine. Please help! 🙏

**Comment by louiebb** [comment]
Friends, I have a tough problem, I am using `Nuxt 3.0.0-rc.6`，`npm install `Install dependencies，
Find `node_module/@nuxt/vite-builder` is version  `Nuxt 3.0.0-rc.11`.  


When I  `npm run dev` has error
```
> nuxt dev -o --mode development
Nuxi 3.0.0-rc.11                                                                                                                  
Nuxt 3.0.0-rc.6                                                                                                            
                                                                                                                                               
  > Local:    http://localhost:3000/
  > Network:  http://10.101.42.55:3000/


 WARN  Module sitemap is disabled due to incompatibility issues:                           
 - [nuxt] Nuxt version `^3.0.0` is required but currently using `3.0.0-rc.6`

✔ Nitro built in 1156 ms                                                                                                  
ℹ Vite client warmed up in 4076ms                                                                                    
[nuxt] [request error] Cannot read property 'startsWith' of undefined
  at ./node_modules/@nuxt/vite-builder/dist/runtime/vite-node-shared.mjs:8:34
  at ModuleJob.run (internal/modules/esm/module_job.js:183:25)
  at async Loader.import (internal/modules/esm/loader.js:178:24)
  at async ./.nuxt/dev/index.mjs:412:26
  at async ./.nuxt/dev/index.mjs:475:64
  at async ./node_modules/h3/dist/index.mjs:592:19
  at async Server.nodeHandler (./node_modules/h3/dist/index.mjs:538:7)
  
```

When I  `npx nuxi@latest upgrade --force`  and  `npm run dev` has error too.

@pi0   help me please !!!!

**Reply by louiebb** [reply]
Is Ok
```
"peerDependencies": {
    "@nuxt/vite-builder": "<=3.0.0-rc.6"
  },
```

**Comment by nyambe** [comment]
I am getting this error when jumping from rc6 to rc11 when npm run dev 
`
[nuxt] [request error] [unhandled] [500] __vite_ssr_import_0__.defuFn is not a function
  at ./.nuxt/app.config.mjs:9:54  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/config.mjs:5:31  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/index.mjs:7:31
[nuxt] [request error] [unhandled] [500] __vite_ssr_import_0__.defuFn is not a function
  at ./.nuxt/app.config.mjs:9:54  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/config.mjs:5:31  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/index.mjs:7:31
[nuxt] [request error] [unhandled] [500] __vite_ssr_import_0__.defuFn is not a function
  at ./.nuxt/app.config.mjs:9:54  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/config.mjs:5:31  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/index.mjs:7:31
`

however npm run build and  npm run generate do work!  I have been able to deploy to amplify with full static.  No api calls. .-)

I get a console.log error however 
`
Uncaught TypeError: Jd.defuFn is not a function
    at entry.fcebda6c.js:formatted:6994:4
(anonymous) @ entry.fcebda6c.js:formatted:6994
`

**Comment by AXYZE9** [comment]
Dear devs, do you already know if theres going to be rc.12 or next release will be final 3.0?

**Reply by thunder87** [reply]
Look at this: https://github.com/nuxt/framework/pull/8067

> This PR includes draft notes for the next release version of Nuxt 3. We plan to release 1-2 more release candidates before the final 3.0.0 version.

**Reply by calebsmithdev** [reply]
Cool - looks like we are getting closer to a launch, but probably still up to 4 weeks out.

**Comment by OleksandrRomaniuk230** [comment]
After updating from v3.0.0-rc.8 to v3.0.0-rc.11

[nuxt] [request error] [unhandled] [500] [vite-node] Failed to load /home/its/Work/projects/baby-games/.nuxt/app.config.mjs
  at ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:180:13)  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/config.mjs:5:31  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)  
  at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (./node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./node_modules/nuxt/dist/app/index.mjs:7:31  
  at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:251:5)
